### PR TITLE
Adjustments to spell facing

### PR DESCRIPTION
**Pull request by chaodhib**
Cherry pick of @r00ty-tc's fix to the issues described here: https://github.com/TrinityCore/TrinityCore/issues/16818

Here is an excerpt of IRC chat:

> [00:26] @r00ty Treeston / chaodhib. https://github.com/r00ty-tc/TrinityCore/commit/c0693a6ee52d31840ab2ee878c4881884b857638 can either of you check that resolves all problems, but still works for spell facing etc (and doesn't cause explosions elsewhere). My only testing is, it compiles and should work in theory :P - I'm away for a week so your other option is to wait that long.
> [00:26] @r00ty Feel free to cherry pick/commit if it's all good, otherwise again it'll be next week ;)

I have tested it (not extensively however) and it looks good.

### CODE REVIEW

**Comment by Nawuko** [comment]
I think with your changes the IsInCombatWith(player) check can be removed. It is not retail behavior.

**Comment by chaodhib** [comment]
@Nawuko Yeah, I agree. Since this is not my code, I prefer to have confirmation before I do the change.

**Comment by r00ty-tc** [comment]
I agree, but that code was already there. Might be worth double checking with treeston to be certain but yes it doesn't make sense to skip players not in combat. 

Also I think the for loop could be turned into a c++11 style one (I forget the feature name) but that's just code style.

**Comment by chaodhib** [comment]
@r00ty-tc In java, it's called a for each loop, not sure for c++.

The change is made. @Treeston any comment?

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -2786,17 +2786,18 @@ bool Creature::FocusTarget(Spell const* focusSpell, WorldObject const* target)
             )
         )
         {
-            const MapRefManager& mapPlayers = GetMap()->GetPlayers();
-            for (MapRefManager::const_iterator it = mapPlayers.begin(); it != mapPlayers.end(); ++it)
-                if (Player* player = (*it).GetSource())
+            std::list<Player*> targets;
+            this->GetPlayerListInGrid(targets, GetVisibilityRange());
```
Comment (Treeston) [inline src/server/game/Entities/Creature/Creature.cpp]: Don't use `this->`

**Comment by Treeston** [comment]
Looks good. :+1:

File: src/server/game/Entities/Creature/Creature.cpp [hunk 2]
```diff
@@ -2786,17 +2786,17 @@ bool Creature::FocusTarget(Spell const* focusSpell, WorldObject const* target)
             )
         )
         {
-            const MapRefManager& mapPlayers = GetMap()->GetPlayers();
-            for (MapRefManager::const_iterator it = mapPlayers.begin(); it != mapPlayers.end(); ++it)
-                if (Player* player = (*it).GetSource())
+            std::list<Player*> playersNearby;
+            GetPlayerListInGrid(playersNearby, GetVisibilityRange());
+            for(Player* player : playersNearby)
```
Comment (MitchesD) [inline src/server/game/Entities/Creature/Creature.cpp]: codestyle for _space_ (

**Comment by r00ty-tc** [comment]
Yep seems fine (at least from what I can see on the phone screen) 

The stray this was there because I used it to find the function name. Then never removed it.

**Pull request merged**